www-apps/webdavcgi: Version bump to 1.1.1

paraenggu opened the pull request:
New version of www-apps/webdavcgi ebuild including the required dev-perl/DateTime-Format-Human-Duration dependency.

Gentoo-Bug: https://bugs.gentoo.org/595060

@dev-zero could you please review and merge? Thanks.

gktrk commented:
@gentoo/proxy-maint

paraenggu commented:
I had a quick chat with @dev-zero it seems that he's quite busy these days and doesn't find the time to review the PR. @gktrk is there someone else from gentoo which could step in and review the pull request?

gktrk commented on www-apps/webdavcgi/webdavcgi-1.1.1.ebuild:
```diff
@@ -0,0 +1,185 @@
+	elog "The WebDAV CGI config is located at ${WEBDAVCONFIG}."
+	elog
+	elog "An example Apache HTTP server configuration snippet is available in"
+	elog "/usr/share/doc/${PF} in the file ${APACHEEXAMPLECONFIG}"
```
Better to use `${ROOT%/}/usr/share/doc/${PF}` here

paraenggu replied:
OK, will change that as well.

gktrk commented:
Are there any security implications to the mode 1777 ?

paraenggu commented on dev-perl/DateTime-Format-Human-Duration/metadata.xml:
```diff
@@ -0,0 +1,19 @@
+	<maintainer type="person">
+		<email>c.affolter@stepping-stone.ch</email>
+		<name>Christian Affolter</name>
+		<description>Maintainer</description>
```
The `<description />` tag is in the [metadata.xml](https://github.com/gentoo/gentoo/blob/master/www-apps/webdavcgi/metadata.xml) of the old webdavcgi ebuild, that's why I added it here as well. Do you want me to remove it?

paraenggu commented on dev-perl/DateTime-Format-Human-Duration/metadata.xml:
```diff
@@ -0,0 +1,19 @@
+	</maintainer>
+	<maintainer type="person">
+		<email>dev-zero@gentoo.org</email>
+		<description>Proxy maintainer</description>
```
Sure, will add the proxy-maint block as well.

gktrk replied:
If we add the proxy-maint block with dev-zero's permission, you don't need the description tags on yourself and him.

paraenggu commented on www-apps/webdavcgi/webdavcgi-1.1.1.ebuild:
```diff
@@ -0,0 +1,185 @@
+HOMEPAGE="http://webdavcgi.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
+
+LICENSE="GPL-3"
```
You're right, I will adapt that accordingly.

paraenggu commented on www-apps/webdavcgi/webdavcgi-1.1.1.ebuild:
```diff
@@ -0,0 +1,185 @@
+
+		export WEBDAVWRAPPERS
+
+	    local wrapper
```
No, it's simply a broken indentation, thanks for spotting it.

paraenggu commented on www-apps/webdavcgi/webdavcgi-1.1.1.ebuild:
```diff
@@ -0,0 +1,185 @@
+	dev-perl/URI
+	dev-perl/UUID-Tiny
+	dev-perl/XML-Simple
+	git? ( dev-vcs/git )
```
Should be correct, according to the [GIT backend documentation](http://webdavcgi.sourceforge.net/doc.html#git), one has to install the git binary.

paraenggu commented:
> Are there any security implications to the mode 1777 ?

There might be security implications if one runs other services on the same system or has system users which have direct access to the file system. They might be able to access temporary or deleted files (in case the thrash feature has been enabled), similar to files within `/tmp`. However, that's the only simple way of having a default out-of-the-box working experience after installing the ebuild. The [ewarn messages](https://github.com/paraenggu/gentoo/blob/webdavcgi-1.1.1/www-apps/webdavcgi/webdavcgi-1.1.1.ebuild#L154-L160) are supposed to inform the user about this circumstances and also provide a hardening hint.

paraenggu commented:
@gktrk Thanks a lot for taking the time to review this PR. Do you prefer new separate commits for addressing the above issues, or shall I squash/rebase the changes with the original commits?

gktrk commented:
I pinged dev-zero for his approval on IRC. We prefer squashing them.

paraenggu commented:
@gktrk I've addressed all your comments and rebased/squashed the changes with the original commits.

gktrk commented:
Merged in https://github.com/gentoo/gentoo/compare/e506143...4c4dd65. Thanks!
